### PR TITLE
Rework Track/TrackSet management

### LINKED ISSUE #61: Rewrite Track handling
As part of #37, I would like to rewrite how the Tracks and Stats work.

This issue is to track (no pun intended) discussions surrounding this topic.

### PR DESCRIPTION
**Pull request by pehala**
* Split Tracks and their Statistics
  * So now the `TrackManager` and `StatsManager` are separate.
  * Add interfaces for `TrackManager` and `StatsManager`.
  * No changes on client-side yet, so the same statistics should show.
* Add `FileSystemTrackManager` and `FileSystemStatsManager`.
  * They are read-only managers that load tracks and their stats.
* Rewrite parsing
  * Works by registering individual LineParsers.
  * A lot of streams used (Can you tell I like functional programming? :D)
* Move track related classes to `shared` module.
* Add tests for the new parser and managers.
* Replace a lot of concrete types with their interfaces (`List` instead of `ArrayList`)
* Change TrackSets format.
  * Previously it used filenames, now it is just names
  * Rewrite existing TrackSets
  * As a feature, it is still broken but it is a little bit less broken than before.

Fixes #61

### CODE REVIEW

File: editor/src/main/java/org/moparforia/editor/EditorPanel.java [hunk 1]
```diff
@@ -227,10 +228,8 @@ public void actionPerformed(ActionEvent e) {
             if(currentTrack == null) {
                 String data = new MapCompressor().compress(canvas.getMap());
                 String name = JOptionPane.showInputDialog(getThis(),"WUTS THE NAME OF THIS TRACK?");
```
[the remark this answers is not in the document]
Comment (pehala) [reply editor/src/main/java/org/moparforia/editor/EditorPanel.java:230]: Editor is a magical place :D

**Comment by pehala** [comment]
I forgot to mention that this PR also removes `morphia` and `mongodb` dependencies, which reduces size of the server binary (4M -> 1,8M)

**Review by PhilippvK — APPROVED** [review]

**Comment by PhilippvK** [comment]
Question: Would it be useful to keep the `CUSTOM` track category or for tracks build with the Editor if they do not fit into any other category?

**Review by PhilippvK — APPROVED** [review]
Code looks very very good. But I was not able to compile and run it yet. I will let you know when I have tested everything.

**Comment by pehala** [comment]
I don't think so, I think all those other categories are enough

**Comment by pehala** [comment]
Editor track can be in normal categories

**Comment by PhilippvK** [comment]
@pehala I will merge this after you rebased `client/client.iml `.

**Comment by pehala** [comment]
Rebased

**Pull request merged**